lib/Elastica/Aggregation/Filters.php

```diff
@@ -22,10 +22,12 @@ class Filters extends AbstractAggregation
      */
     public function addFilter(AbstractFilter $filter, $name = '')
     {
-        if (empty($name)) {
-            $filterArray[] = $filter->toArray();
+        $filterArray = array();
+
+        if (is_string($name)) {
```

Comment:
@ewgRa Not sure what you meant with your second last sentence. Please open a pull request with some tests if possible for the suggested changes, so we can directly discuss it based on code.